### ISSUE TITLE
Swap disk/file 

### ISSUE THREAD

**Issue by sergelogvinov** [siderolabs/talos#4440]
## Feature Request

Make possible to add swap disk/file to the system at boot time.

### Description

When memory is not enough or memory pressure, system processes need a few extra memory to finish some jobs/gracefully stop pods etc. A small swap disk can help with that.

Proposal:

```yaml
# example
machine:
  swap:
     disk: /dev/sda
     # or
     size: 1Gb
```

The swap can be disk/partition (azure has special disk for swap) or file in ephemeral storage.
Swap has to be alway encrypted. The encryption key is auto generated at boot time.
Sysctl param sets to ```vm.swappiness=1```

If ```disk``` param is not set - Talos creates a swap-file with defined ```size```.

```disk``` and ```size``` cannot be set in the same time(?)

### Community Note

* Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request

**Comment by andrewrynhard** [comment]
It is explicitly called out in the Kubernetes docs to disable swap.

**Comment by ammmze** [comment]
[v1.22 brings alpha support for swap](https://kubernetes.io/blog/2021/08/09/run-nodes-with-swap-alpha/)...so I think there is still some validity to this request.

**Comment by smira** [comment]
# Notes from planning meeting

:+1:

**Comment by abckey** [comment]
any update on this ?

**Comment by smira** [comment]
> any update on this ?

no plans for it so far, even though it should be relatively trivial to implement

**Comment by Max-Sum** [comment]
Are we getting this feature soon?

**Comment by samos667** [comment]
https://kubernetes.io/blog/2023/08/24/swap-linux-beta/

It would be really great to be able to do that, with the possibility of using ZRAM in the best of all worlds.

**Comment by stevefan1999-personal** [comment]
This is also now needed for specific scenario of using budget VPSes to deploy Talos, even just as a worker, #9037 is a particular case where image pulling can cause OOM and eventually kill containerd, making the whole installation process stuck

**Comment by pio2398** [comment]
Still relevant

**Comment by andrewchen5678** [comment]
Please implement this if it is easy to implement, there are valid use cases to use swap space on environments like vps where memory is limited.

**Comment by stevefan1999-personal** [comment]
@andrewchen5678 Yep! This is especially crucial for environments with memory constraints such as on a NUC, so we usually enable ZRam to get some memory compressed, trading memory with computational power in exchange.

**Comment by np22-jpg** [comment]
If someone wanted to attempt this, where would they start? I’ve seen the examples in in [siderolabs/extensions](https://github.com/siderolabs/extensions), but I believe enabling zram requires changes to the [kubelet configuration](https://kubernetes.io/blog/2023/08/24/swap-linux-beta/#set-up-a-kubernetes-cluster-that-uses-swap-enabled-nodes), which I’m not sure about approaching.

**Comment by smira** [comment]
We are still looking towards enabling this via Volume Management. This on the radar, but I can't promise any exact date.

**Comment by frezbo** [comment]
Fixed by https://github.com/siderolabs/talos/issues/10674

**Comment by septatrix** [comment]
Should we create a follow up issue for zram?

**Comment by smira** [comment]
> Should we create a follow up issue for zram?

See https://github.com/siderolabs/talos/discussions/11308
